src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { peopleFromServer } from '../data/people';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+type Props = {
+  onSelected?: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [list, setList] = useState<boolean>(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
The `applyQuery` function should include `delay` as a dependency in the dependency array of `useCallback`. This ensures that if `delay` changes, the debounced function is updated accordingly.